### PR TITLE
[Frontend/Inputs] Create generic input components

### LINKED ISSUE #15: [Frontend] Create generic input components
Create generic input component including :
- [x] Styling
- [x] Responsive
- [x] Error state handling
- [x] Text, Number, Select, Checkboxes
- [x] Two-way binding value
- [x] Tests

### PR DESCRIPTION
**Pull request by pchatard**
Resolves #15.

Create basic (fully functional) styled inputs with error states :

- Text, number, URL inputs with optional copy option
- Checkbox / Toggle button
- Select with optional search option
- Basic Date picker
- Unit tests for each component
- Added potential improvements in README.md

### CODE REVIEW

File: .github/workflows/build_front.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# This workflow will do a clean install of node dependencies, cache/restore them, build the source code and run tests across different versions of node
```
Comment (ndureiss) [inline .github/workflows/build_front.yml:1]: Rename to build&test_front to suit content

File: .github/workflows/build_front.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Build Frontend
```
Comment (ndureiss) [inline .github/workflows/build_front.yml]: Here too

File: .github/workflows/build_front.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    branches: 
+      - master
+      - dev
+      - v1
+  pull_request:
+    branches: 
+      - "*"
+
+jobs:
+  build:
```
Comment (ndureiss) [inline .github/workflows/build_front.yml]: try to separate jobs ? build / lint / test ?

File: webapp-giftlist-1.0/README.md [hunk 4]
```diff
@@ -84,6 +84,28 @@
...
+    - [x] ~~Erreur~~
+    - [x] ~~Responsive~~
+- [x] ~~Gérer le contenu avec le state~~
+- [x] ~~Gérer les erreurs~~
+- [x] ~~Option "Copy to clipboard"~~
+- [x] ~~Ecrire des tests~~
+- [ ] Ajouter un Date Picker custom
+- [ ] Ajouter un bouton Ouvrir sur l'input URL
+- [ ] Option Icon
+- [ ] Ajouter un bouton de reset
```
Comment (ndureiss) [inline webapp-giftlist-1.0/README.md:107]: Not considered ? or just not updated ?

File: .github/workflows/build_front.yml (hunk 3 above)
Comment (ndureiss) [reply .github/workflows/build_front.yml]: be careful, jobs name must be not redundant between other actions file (for better understanding)

File: webapp-giftlist-1.0/src/components/Inputs/FormInputLink.vue [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+
+		const computedLabelClass = computed(() => {
+			if (props.isError) {
+				return "text-red-600";
+			}
+			if (selected.value) {
+				return "text-indigo-600";
+			}
+			return "text-gray-600";
+		});
```
Comment (ndureiss) [inline webapp-giftlist-1.0/src/components/Inputs/FormInputLink.vue]: seems redundant, can be mutualized ?

File: webapp-giftlist-1.0/src/components/Inputs/FormInputLink.vue [hunk 6]
```diff
@@ -0,0 +1,189 @@
+<template>
+	<fieldset class="flex flex-col items-start transition-all">
+		<label class="text-sm font-semibold transition-all" :class="computedLabelClass">
+			{{ label }}
+		</label>
+		<div
+			class="input-container flex items-stretch overflow-hidden border rounded-md my-1 w-full"
+			:class="computedInputBorderClass"
+		>
```
Comment (ndureiss) [inline webapp-giftlist-1.0/src/components/Inputs/FormInputLink.vue]: seems redundant, can't be mutualized ?

**Review by ndureiss — COMMENTED** [review]
Very good job! What does it look like ? Few remarks, rest LGTM!

File: .github/workflows/build_front.yml (hunk 1 above)
Comment (pchatard) [reply .github/workflows/build_front.yml:1]: Sure, will change. It was a first test, I'm not really used to all of this. We'll improve this later !

File: webapp-giftlist-1.0/README.md (hunk 4 above)
Comment (pchatard) [reply webapp-giftlist-1.0/README.md:107]: Simply not done yet, because these are not necessary for the components to be functional

**Pull request merged**